CMakeScripts/OCInstrumentation.cmake

```diff
@@ -0,0 +1,34 @@
+
+SET(OC_INSTRUMENTATION none)
+if (INSTRUMENTATION)
+    STRING(TOLOWER "${INSTRUMENTATION}" INSTRUMENTATION)
+    if (INSTRUMENTATION STREQUAL "scorep")
+        SET(OC_INSTRUMENTATION scorep)
+	# Reset compiler names to their scorep wrapper compiler name
+	if(DEFINED CMAKE_C_COMPILER)
+	  string(CONCAT CMAKE_C_COMPILER "scorep-" "${CMAKE_C_COMPILER}")
+	endif()
+	if(DEFINED CMAKE_CXX_COMPILER)
+	  string(CONCAT CMAKE_CXX_COMPILER "scorep-" "${CMAKE_CXX_COMPILER}")
+	endif()
+	if(DEFINED CMAKE_Fortran_COMPILER)
+	  string(CONCAT CMAKE_Fortran_COMPILER "scorep-" "${CMAKE_Fortran_COMPILER}")
+	endif()
+    elseif (INSTRUMENTATION STREQUAL "vtune")
+        if(TOOLCHAIN STREQUAL "intel")
+          SET(OC_INSTRUMENTATION vtune)
+	else()
+          message(WARNING "Can only use vtune instrumentation with an Intel toolchain. Proceeding with no instrumentation.")
+        endif()
+    elseif (INSTRUMENTATION STREQUAL "none")
+        # Do nothing
+
+    else()
+        message(WARNING "Unknown instrumentation: ${INSTRUMENTATION}. Proceeding with no instrumentation.")
+    endif()
+endif()
+
+message(STATUS "DEBUG: OC_INSTRUMENTATION     = ${OC_INSTRUMENTATION}")
```

Comment:
I've commented these out. However, what we should have is a better solution to debugging CMake. I would advocate having a CMakeScript (OCUtilities.cmake or OCDebug.cmake or something) that doesn't rely on to many OC variables so that it can be included at the top of all of the CMake scripts that we have. We could then have something like in this file

function (ocdebug message)
    if(DEFINED OC_DEBUG_CMAKE)
         message(STATUS "DEBUG: ${message}" 
    endif()
endfunction()

Then, in all the CMake scripts we would have something like

include(OCDebug.cmake)

and

ocdebug("Debug output information")

This would unify all the messaged and messagev functions scattered throughout the scripts. We could then leave them uncommented as they wouldn't produce output unless somebody did a 

cmake -DOC_DEBUG_CMAKE=1

or something.